docs/cookbook/database-doctrine.md

```diff
@@ -0,0 +1,200 @@
+---
+title: Using Doctrine with Slim
+---
+
+This cookbook entry describes how to integrate from scratch the widely used [Doctrine ORM](http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/) into a Slim application.
+
+## Adding Doctrine to your application
+
+The first step is importing the library into the `vendor` directory of your project using [composer](https://getcomposer.org/).
+
+<figure>
+{% highlight bash %}
+composer require doctrine/orm:^2.5
+{% endhighlight %}
+<figcaption>Figure 1: Require doctrine in your application.</figcaption>
+</figure>
+
+## Provide database credentials
+
+Next, add the database settings alongside your Slim configuration.
+
+<figure>
+{% highlight php %}
+<?php
+
+return [
+    'settings' => [
+        'determineRouteBeforeAppMiddleware' => false,
+        'displayErrorDetails' => true,
+
+        'doctrine' => [
+            // list all the paths were you keep your annotated entities
+            'paths = [__DIR__ . '/../src/Domain/Entity']
+            'conn' => [
+                'driver' => 'pdo_mysql',
+                'host' => 'localhost',
+                'port' => 3306,
+                'dbname' => 'mydb',
+                'user' => 'user',
+                'password' => 'secret',
+                'charset' => 'utf-8'
+            ]
+        ]
+    ],
+];
+{% endhighlight %}
+<figcaption>Figure 2: Slim settings array.</figcaption>
+</figure>
+
+## Define the EntityManager service
+
+Now we define the `EntityManager` service, which is the primary way to interact with Doctrine. Here we show how to configure the metadata reader to work with PHP annotations, which is at the same time the most used mode and the most tricky to set up.
+
+Alternatively XML or YAML can also be used to describe the database schema.
+
+<figure>
+{% highlight php %}
+
+use Doctrine\Common\Annotations\AnnotationReader;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\Mapping\Driver\AnnotationDriver;
+use Doctrine\ORM\Tools\Setup;
+
+$container[EntityManager::class] = function ($container) {
+    $config = Setup::createAnnotationMetadataConfiguration(
+        $container['settings']['doctrine']['paths'], true
+    );
+    
+    $config->setMetadataDriverImpl(
+        new AnnotationDriver(
+            new AnnotationReader,
+            $container['settings']['doctrine']['paths']
+        )
+    );
+
+    return EntityManager::create(
+        $container['settings']['doctrine']['conn'], $config
+    );
+};
+{% endhighlight %}
+<figcaption>Figure 3: Defining the EntityManager service.</figcaption>
+</figure>
+
+## Create the Doctrine console
+
+To be able to automatically run database migrations, validate class annotations and so on you need to create the Doctrine CLI application.
```

Comment:
The console should be placed in the root of the project in the `cli-config.php` file (as per standard doctrine)